src/doc/src/reference/specifying-dependencies.md

```diff
@@ -454,8 +454,51 @@ following to the above manifest:
 log-debug = ['bar/log-debug'] # using 'foo/log-debug' would be an error!
 ```
 
+### Inheriting a dependency from a workspace
+
+Dependencies can be inherited from a workspace by specifying the
+dependency in the workspace's [`[workspace.dependencies]`][workspace.dependencies] table.
+After that add it to the `[dependencies]` table with `dep.workspace = true`.
+
+The `workspace` key can be defined with:
+- [`optional`][optional]: Note that the`[workspace.dependencies]` table is not allowed to specify `optional`.
+- [`features`][features]: These are additive with the features declared in the `[workspace.dependencies]`
+
+The `workspace` key cannot be defined with:
+
+|                  |                    | 
+|------------------|--------------------|
+| `branch`         | `default-features` |
+| `git`            | `package`          |
+| `path`           | `registry`         |
+| `registry-index` | `rev`              |
+| `tag`            | `version`          |
```

Comment:
I'm a little concerned with trying to keep this list up-to-date over time, and in general isn't necessary for the user to know exactly which keys aren't allowed. Perhaps this could be simplified with the following?

```suggestion
Dependencies with the `workspace=true` key cannot use any other dependency key (such as `version` or `default-features`).
```

Reply:
Do you think I should remove the table in `workspaces.md` as well?

Reply:
I assume this was meant to be here
https://github.com/rust-lang/cargo/pull/10859#discussion_r924002389
> I think it's fine to keep that table, as it is important for the user to know which keys they can use. We'll just need to remember to update it when any new inherited keys are added.